compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -595,10 +595,15 @@ void OMR::CodeGenerator::lowerTrees()
 
 void OMR::CodeGenerator::postLowerTrees()
    {
-   if (comp()->getOption(TR_SplitWarmAndColdBlocks) &&
-       !comp()->compileRelocatableCode())
+   TR::Compilation *comp = self()->comp();
+
+   bool moveMethodToColdCode = comp->getOption(TR_MoveColdMethodsToColdCode) &&
+                               (comp->getOptLevel() == cold || comp->getOptLevel() == reducedWarm);
+
+   if ((comp->getOption(TR_SplitWarmAndColdBlocks) || moveMethodToColdCode) &&
+       !comp->compileRelocatableCode())
       {
-      self()->prepareLastWarmBlockForCodeSplitting();
+      self()->prepareLastWarmBlockForCodeSplitting(moveMethodToColdCode);
```

Comment:
Why don't we move the AOT code entirely to the cold area?

Reply:
Also, if we move the entire method to the cold area, do we still need to prepareLastWarmBlockforCodeSplitting?

Reply:
I would suggest to first see if it helps footprint and then improve if it does.

Reply:
What was the intent behind `comp->getOptLevel() == reducedWarm` ? This is an opt level that we do not use currently. There is the notion another notion of reduced-warm, but that applies to warm compilation without optServer. E.g.: 
`bool isColdOrReducedWarm = (comp->getMethodHotness() < warm) || (comp->getMethodHotness() == warm && comp->getOption(TR_NoOptServer));`

Reply:
Good to know. Thanks.